lib/ledger_sync/adaptors/quickbooks_online/webhook.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative 'webhook_event'
+
+# ref: https://developer.intuit.com/app/developer/qbo/docs/develop/webhooks/managing-webhooks-notifications#validating-the-notification
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      class Webhook
+        attr_reader :notifications,
+                    :original_payload,
+                    :payload
+
+        def initialize(payload:)
+          @original_payload = payload
+          payload = JSON.parse(payload) if payload.is_a?(String)
+          @payload = payload
+
+          event_notifications_payload = payload.dig('eventNotifications')
+          raise 'Invalid payload: Could not find eventNotifications' if event_notifications_payload.blank?
```

Comment:
I'd rather not raise an error and just return empty resources/notifications. This seems like risky behavior since QBO is hard to trust

Reply:
True.  But I actually just intended for this to check for an array.  It will error out below because it can't parse anything if not.  I figured this error was friendlier than chasing down why it was erroring in the first place.